jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/JavaContainerBuilderHelper.java

```diff
@@ -53,8 +53,12 @@ public static LayerConfiguration extraDirectoryLayerConfiguration(
             localPath -> {
               AbsoluteUnixPath pathOnContainer =
                   AbsoluteUnixPath.get("/").resolve(extraDirectory.relativize(localPath));
-              builder.addEntry(
-                  localPath, pathOnContainer, extraDirectoryPermissions.get(pathOnContainer));
+              FilePermissions permissions = extraDirectoryPermissions.get(pathOnContainer);
```

Comment:
See if it is trivial to make this return `Optional` in this PR.

Reply:
Do you mean this line should be `Optional<FilePermissions> permissions = ...`, or do you mean something else?

Reply:
Yeah, sorry, I meant `Optional<FilePermissions> permissions = ...`.

Reply:
Never mind. I didn't realize this is just a map.